docs/architecture/cluster-registration.md

```diff
@@ -0,0 +1,25 @@
+# Cluster registry workflow
+
+The basic registration process is that the agent first sends a registration request to the hub cluster through the bootstrap token of the hub cluster. After the hub cluster detects the registration request, the hub cluster will start to create some necessary resources for registration. The registration request will be marked as approved after these resources to be created and ready.  When the cluster registration success, the agent will periodically send its heartbeat information to the hub cluster.
+
+## Architecture
+
+<img src="../images/cluster-registry-workflow.png" style="width:1000px;"/>
+
+## Workflow
+
+1. [Agent] Check if the `parent-cluster` exists.
+2. [Agent] Generate parent dedicated kubconfig
+   1. If the `parent-cluster` exists, use it.
+   2. If the `parent-cluster` does not exist, use bootstrap token.
+3. [Agent] Create `ClusterRegistrationRequest` with `step 2` kubeconfig, and wait for the request to be approved.
+4. [Hub] CRR controller watched the CRR created by [Agent], create necessary objects `ManagedCluster`, `Dedicated Namespace` and default rbac rules.
+5. [Hub] Get default parent cluster credential from `step 4` for child cluster access parent cluster.
+6. [Hub] Approve & Update CRR status:
+   1. Results, if all above `steps[4-5]` have been created and ready, result will be approved.
+   2. DedicatedToke, from step 6 generated secret.
+   3. CACertificate, from step 6 generated secret.
+7. [Agent] Get parent cluster credential from CRR status.
+8. [Agent] Save parent cluster credential to `parent-cluster` secret.
+
+End here, the child cluster is registered. The child cluster will report status to the hub with statusReportFrequency.
```

Comment:
After the registration, the statusManager will report the child cluster status to the hub as defined in `statusReportFrequency`.